src/databricks/sql/utils.py

```diff
@@ -387,6 +388,36 @@ def inject_parameters(operation: str, parameters: Dict[str, str]):
     return operation % parameters
 
 
+def transform_paramstyle(operation: str) -> str:
+    """
+    Performs a regex substitution such that any occurence of `%(param)s` will be replaced with `:param`
+
+    This utility function is built to assist users in the transition between the default paramstyle in
+    this connector prior to version 3.0.0 (`pyformat`) and the new default paramstyle (`named`).
+
+    Args:
+        operation (str): The operation or SQL text to transform.
+
+    Returns:
+        str
+    """
+
+    _output_operation = operation
+
+    PYFORMAT_PARAMSTYLE_REGEX = r"%\((\w+)\)s"
```

Comment:
do we have an existing regex for inline replacement today, can we reuse that logic?

Reply:
Nope. We're just doing string interpolation so Python handles it for us, in C.

Have a look at `inject_parameters` to see exactly what it does:

https://github.com/databricks/databricks-sql-python/blob/0e7ce57ae4fecc364c4de55d6adace33c39aff5f/src/databricks/sql/utils.py#L386-L387

Reply:
can we call to inject_parameters which replace parameters with the native format of parameter

Reply:
That's what `transform_paramstyle()` already does...